rebasehelper/utils.py

```diff
@@ -138,6 +142,101 @@ def cprint(cls, message, fg=None, bg=None, style=None):
         else:
             print(message)
 
+    @staticmethod
+    def color_is_light(rgb):
+        """Function to determine whether a color is dark or light.
+
+        Args:
+            rgb (tuple): RGB tuple with last element specifying number of RGB digits.
+
+        Returns:
+            bool: Whether color is light or dark.
+
+        """
+        brightness = 1 - (0.299 * rgb[0] + 0.587 * rgb[1] + 0.114 * rgb[2]) / (16 ** rgb[3] - 1)
+        return brightness < 0.5
+
+    @staticmethod
+    def parse_rgb_device_specification(specification):
+        """Parses RGB device specification.
+
+        Args:
+            specification (str): RGB device specification.
+
+        Returns:
+            tuple: RGB tuple with the last element being number of hex digits it was parsed from.
+
+        """
+        specification = re.sub(r'rgb:', '', specification)
+        rgb = specification.split('/')
+        return tuple(int(x, 16) for x in rgb) + (len(rgb[0]),)
+
+    @staticmethod
+    def dark_or_light_background():
+        """Gets terminal background color and decides whether it is light or dark.
+
+        Returns:
+            str: Whether to use dark or light color scheme.
+
+        """
+        background_color = ConsoleHelper.exchange_control_sequence('\x1b]11;?\x07')
+
+        if not re.match(r'^rgb:([A-Fa-f0-9]{1,4})/([A-Fa-f0-9]{1,4})/([A-Fa-f0-9]{1,4})$', str(background_color)):
```

Comment:
WTF is this? First you use regexp to parse the specification, then you just forget about it and use a different (and much less robust) parsing method.